### PR TITLE
Memset on typed specialized values

### PR DESCRIPTION
**Pull request by Cellule**
Sometimes, when type specializing the original instruction is lost.
ie: `s3.var          =  LdC_A_I4       0 (0x0).i32` is type specialized to `int32`, but the dst is used as a float. 
If it's the only use, then `s3.var` is removed.
Previously memset was trying to get `s3.var` from the float type specialized version. 
However, it is not live anymore.

This change makes memset use directly the value the `StElemI_A` is using (ie: the float typed specialized register)
And will call ToVar on it.
A better fix would be to try to use the var version first if it is available, but might introduce unexpected bugs.

I also added several test cases to cover this issue and more, along with some tests with simds.

[Review on Reviewable](https://reviewable.io/reviews/microsoft/chakracore/219)

### CODE REVIEW

File: test/Array/memset_tester.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
+function* makeValueGen(a, b) {
```
Comment (dilijev) [inline test/Array/memset_tester.js]: Missing copyright.

File: lib/Backend/GlobOpt.cpp [hunk 2]
```diff
@@ -4254,17 +4254,8 @@ GlobOpt::CollectMemsetStElementI(IR::Instr *instr, Loop *loop)
...
-            StackSym* sym = opnd->GetStackSym();
-            if (sym->GetType() != TyVar)
-            {
-                varSym = sym->GetVarEquivSym(instr->m_func);
-            }
-            else
-            {
-                varSym = sym;
-            }
+            varSym = opnd->GetStackSym();
```
Comment (rajatd) [inline lib/Backend/GlobOpt.cpp]: nit: might want to rename this `varSym` to avoid confusion

**Comment by rajatd** [comment]
Looks good overall

**Comment by LouisLaf** [comment]
LGTM.

File: lib/Backend/GlobOpt.cpp (hunk 2 above)
Comment (Cellule) [inline lib/Backend/GlobOpt.cpp]: Agreed, I wanted to make the change, but I forgot about it :)

File: test/Array/memset_tester.js (hunk 1 above)
Comment (Cellule) [inline test/Array/memset_tester.js]: I added the copyright

File: lib/Backend/GlobOpt.cpp (hunk 2 above)
Comment (Cellule) [inline lib/Backend/GlobOpt.cpp]: I renamed it to `srcSym`. I also renamed the field in `MemSetCandidate`

**Pull request merged**